src/Interpreters/JoinInfo.h

```diff
@@ -134,6 +148,30 @@ struct JoinCondition
     /// Unlike the join predicates, these conditions can be arbitrary expressions.
     std::vector<JoinActionRef> residual_conditions;
 
+    void fixReferences(const JoinExpressionActions & expression_actions)
```

Comment:
Let's move it into .cpp

Maybe better to make it a `static JoinCondition clone(const JoinExpressionActions &)`